ocs_ci/utility/prometheus.py

```diff
@@ -0,0 +1,118 @@
+import base64
+import errno
+import logging
+import os
+import requests
+import tempfile
+import yaml
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.ocs.ocp import OCP
+
+logger = logging.getLogger(name=__file__)
+
+
+class PrometheusAPI(object):
+    """
+    This is wrapper class for Prometheus API.
+    """
+
+    _token = None
+    _user = None
+    _password = None
+    _endpoint = None
+    _cacert = None
+
+    def __init__(self, user=None, password=None):
+        """
+        Constructor for PrometheusAPI class.
+
+        Args:
+            user (str): OpenShift username used to connect to API
+        """
+        self._user = user or config.RUN['username']
+        if not password:
+            filename = os.path.join(
+                config.ENV_DATA['cluster_path'],
+                config.RUN['password_location']
+            )
+            if os.path.exists(filename):
+                with open(filename) as f:
+                    password = f.read()
+            else:
+                raise FileNotFoundError(
+                    errno.ENOENT,
+                    os.strerror(errno.ENOENT),
+                    filename
+                )
```

Comment:
You don't need to test existence of the file, if the only thing you are going to do is to open it and raise `FileNotFoundError` exception if the file doesn't exist manually. Is there any reason why you can't just open and read the file like shown below without the if - else - raise wrapper code?

```
                with open(filename) as f:
                    password = f.read()
```

Reply:
See:

```
$ ipython3 
Python 3.7.3 (default, May 11 2019, 00:38:04) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.2.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: with open("nenineni") as f: 
   ...:     print(f.read()) 
   ...:                  
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-1-1afc67e7ddb6> in <module>
----> 1 with open("nenineni") as f:
      2     print(f.read())
      3 

FileNotFoundError: [Errno 2] No such file or directory: 'nenineni'
```

Reply:
Good point. Fixed.